UnitedIncome/components
# Storybook DocsPage/MDX Support

### Description
> Please provide a brief description of your changes below.

This branch introduces a proof of concept for Storybook's new DocsPage update which you can view [here](https://medium.com/storybookjs/storybook-docspage-e185bc3622bf).

<img width="990" alt="Screen Shot 2019-10-08 at 10 07 10 AM" src="https://user-images.githubusercontent.com/10888441/66413563-9cccb000-e9c5-11e9-8ed9-ade07b0024b5.png">

There's a few big changes in here.

* `Component.story.js` and `Component.md` are now merged into one, and now have a single unified `Component.story.mdx` file. These files define the documentation and the stories in one file.
* `addon-info` has  been removed as it's now obsolete. `addon-readme` will soon follow once all stories have been converted.
* Stories are defined inline using a set of provided components by Storybook. You can view a preview of what this looks like here with the `<Box />` component.
* Prop tables are generated using `<Props for={Component} />` inside the `mdx` file.
* I've updated the linting rules, along with the generator to support `mdx`.\
* Added some additional UI colors to the theme.

In this pull request I've updated the `Box`, `BubbleSelector` and the `BubbleDropdown` components to support this new format. To view the changes you can click on the story in Storybook and click on the `Docs` tab at the top to view the new examples.

#### Known Issues

There's a couple of known issues with this.

* Using addon-knobs with a helper function to generate props results in the actual function call being rendered in the code preview instead of the computed values. I've placed an issue with the Storybook team on this here: https://github.com/storybookjs/storybook/issues/8342
* For whatever `forceReRender` isn't needed anymore for stateful components. Not sure if there will be any other side effects from that later down the line or not.

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* Load up a story that is supporting this new format and click the `Docs` tab. This tab will show for any story.
* All old stories using the old format of `Component.story.js` will continue to function, but should no longer be used.

### Comments
> Any additional comments that will aid with the review process.

I'd like to update all of the old stories into this format, and I predict it will take around an afternoon to complete this work if everyone is on board. I think this way of writing documentation is much cleaner.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-4006)
